syne_tune/experiments.py

```diff
@@ -93,20 +153,35 @@ def metric_names(self) -> List[str]:
     def entrypoint_name(self) -> str:
         return self.metadata["entrypoint"]
 
-    def best_config(self) -> Dict[str, Any]:
+    def best_config(self, metric: Union[str, int] = 0) -> Dict[str, Any]:
         """
-        Return the best config found for the first metric defined in the scheduler.
+        Return the best config found for the specified metric
+        :param metric: Indicates which metric to use, can be the index or a name of the metric.
+                       default to 0 - first metric defined in the Scheduler
         :return: Configuration corresponding to best metric value
         """
-        metric_names = self.metric_names()
-        metric_mode = self.metric_mode()
-
-        if len(metric_names) > 1:
-            logging.warning(
-                "Several metrics exists so the best is not defined, this will "
-                f"return the best w.r.t. the first metric of {metric_names}."
+        if isinstance(metric, str):
```

Comment:
This codeblock (allowing `metric` to be str or int, and map to index and name) looks more generally useful. Method?

Reply:
Including determining `metric_mode`

Reply:
sure I can extract that